packages/eslint-plugin/src/rules/require-localize-metadata.ts

```diff
@@ -51,23 +90,51 @@ export default createESLintRule<Options, MessageIds>({
             type: 'boolean',
             default: DEFAULT_OPTIONS.requireMeaning,
           },
+          requireCustomId: {
+            type: 'boolean',
+            default: DEFAULT_OPTIONS.requireCustomId,
+          },
+          requireDefaultValue: {
+            type: 'boolean',
+            default: DEFAULT_OPTIONS.requireDefaultValue,
+          },
+          boundTextAllowedPattern: {
```

Comment:
I don't think this is very clear that it relates to custom ID.

I think something like this would be much clearer:

```json
{
  "@angular-eslint/require-localizae-metadata": [
     "error",
     {
       "requireCustomId": true,
       "requireCustomIdMatchingPattern": "custom_id"
     }
  ]
}
```

We potentially could then take it a step further and dynamically default `requireCustomId` to true when `requireCustomIdMatchingPattern`?

Reply:
With that said, I am not super clear on the examples you have of:

```
boundTextAllowedPattern:
          '^(?<lib>.[^\\.]*)\\.(?<component>.[^\\.]*)\\.(?<tag>.[^\\.]*)\\.(?<attribute>.[^\\.]*)?\\.(?<name>.[^\\.]*).$',
```

Are all of the sections potentially applicable to just the custom ID in practice? (I am not somebody who has ever leveraged the i18n feature personally). Maybe you could please give some real world examples?

Reply:
To answer your last question:

This is an example of pattern on which you have a more verbose pattern that can be used for more people to understand where the translation is used and that can help to identify and avoid duplicated ids.

Here you have an example of a custom id that matches this regular expression: 'cards-management.form-details.input.placeholder.title'

This is just an example and can be overriden, but I thought it could be interesting to have an example.

Reply:
> With that said, I am not super clear on the examples you have of:
> 
> ```
> boundTextAllowedPattern:
>           '^(?<lib>.[^\\.]*)\\.(?<component>.[^\\.]*)\\.(?<tag>.[^\\.]*)\\.(?<attribute>.[^\\.]*)?\\.(?<name>.[^\\.]*).$',
> ```
> 
> Are all of the sections potentially applicable to just the custom ID in practice? (I am not somebody who has ever leveraged the i18n feature personally). Maybe you could please give some real world examples?

I thought that it could be interesting to give some example to the developers to avoid duplicated custom ids and also to provide some more information about where the custom id is used.

Here you have an example of a custom id matching the regular expression: "cards-management.card-details-form.input.placeholder.remaining-credit"

Reply:
> I don't think this is very clear that it relates to custom ID.
> 
> I think something like this would be much clearer:
> 
> ```json
> {
>   "@angular-eslint/require-localizae-metadata": [
>      "error",
>      {
>        "requireCustomId": true,
>        "requireCustomIdMatchingPattern": "custom_id"
>      }
>   ]
> }
> ```
> 
> We potentially could then take it a step further and dynamically default `requireCustomId` to true when `requireCustomIdMatchingPattern`?

I have changed the name of the property to the one you have suggested.

How would you like to dynamically default `requireCustomId` to true?

Reply:
You changed the messageId but not the option which is the more prominent user-facing thing that I was concerned about.

For a dynamic default, you could just not specify it on the schema and apply it within the rule implementation at runtime, right?